### ISSUE TITLE
Reverse-proxy configuration changes break my setup; docker: Invalid ip address 127.0.0.1:: address 127.0.0.1::: too many colons in address.

### ISSUE THREAD

**Issue by schwony** [spantaleev/matrix-docker-ansible-deploy#2476]
**Describe the bug**
With the new role `matrix_playbook_reverse_proxy_type: other-nginx-non-container`, only secondary containers are started.
<img width="1305" alt="Bildschirm­foto 2023-02-13 um 14 09 25" src="https://user-images.githubusercontent.com/63668511/218466873-aa6cfeb0-b3f1-473d-8518-036d45aedea8.png">

matrix-synapse-reverse-proxy-companion.service & matrix-ma1sd.service fail to start with 
**`docker: Invalid ip address 127.0.0.1:: address 127.0.0.1::: too many colons in address.`**

this might be related to issue #2474 

**To Reproduce**
My `vars.yml` file looks like this:

```
matrix_playbook_reverse_proxy_type: 'other-nginx-non-container'
matrix_etherpad_enabled: true
matrix_etherpad_admin_username: <redacted>
matrix_etherpad_admin_password: <redacted>
matrix_mautrix_whatsapp_enabled: true
matrix_synapse_configuration_extension_yaml: |
  experimental_features:
    msc2716_enabled: true
matrix_mautrix_whatsapp_configuration_extension_yaml:
  bridge:
    history_sync:
      backfill: true
matrix_mautrix_telegram_enabled: true
matrix_mautrix_telegram_api_id: <redacted>
matrix_mautrix_telegram_api_hash: <redacted>
matrix_homeserver_generic_secret_key: <redacted>
matrix_mailer_sender_address: <redacted>
matrix_mailer_relay_use: true
matrix_mailer_relay_host_name: <redacted>
matrix_mailer_relay_host_port: 587
matrix_mailer_relay_auth: true
matrix_mailer_relay_auth_username: <redacted>
matrix_mailer_relay_auth_password: <redacted>
matrix_well_known_matrix_server_enabled: true
matrix_ma1sd_enabled: true
matrix_synapse_enable_registration: true
matrix_synapse_registrations_require_3pid: 
  - 'email'
matrix_ma1sd_configuration_extension_yaml: |
  register:
    policy:
      allowed: true
      threepid:
        email:
          domain:
            blacklist: ~
            whitelist: ~
  matrix_ma1sd_threepid_medium_email_identity_from: <redacted>
  matrix_ma1sd_threepid_medium_email_connectors_smtp_host: <redacted>
  matrix_ma1sd_threepid_medium_email_connectors_smtp_port: 587
  matrix_ma1sd_threepid_medium_email_connectors_smtp_tls: 1
  matrix_ma1sd_threepid_medium_email_connectors_smtp_login: <redacted>
  matrix_ma1sd_threepid_medium_email_connectors_smtp_password: <redacted>
matrix_nginx_proxy_proxy_matrix_3pid_registration_enabled: true
matrix_nginx_proxy_proxy_matrix_nginx_status_enabled: false
devture_postgres_backup_enabled: true
devture_postgres_backup_schedule: '@daily'
devture_postgres_backup_keep_days: 7
devture_postgres_backup_keep_weeks: 4
devture_postgres_backup_keep_months: 12
devture_postgres_backup_base_path: "{{ matrix_base_data_path }}/postgres-backup"
matrix_prometheus_enabled: true
prometheus_node_exporter_enabled: true
matrix_grafana_enabled: true
matrix_grafana_anonymous_access: false
matrix_grafana_default_admin_user: <redacted>
matrix_grafana_default_admin_password: <redacted>
matrix_etherpad_enabled: false
matrix_mautrix_signal_enabled: true
devture_postgres_connection_password: <redacted>
matrix_synapse_ext_password_provider_rest_auth_enabled: true
matrix_synapse_ext_password_provider_rest_auth_endpoint: "http://matrix-ma1sd:8090"
matrix_synapse_ext_password_provider_rest_auth_registration_enforce_lowercase: false
matrix_synapse_ext_password_provider_rest_auth_registration_profile_name_autofill: true
matrix_synapse_ext_password_provider_rest_auth_login_profile_name_autofill: false
matrix_synapse_ext_password_provider_shared_secret_auth_enabled: true
matrix_synapse_ext_password_provider_shared_secret_auth_shared_secret: <redacted>
matrix_synapse_admin_enabled: true
#matrix_docker_installation_enabled: true
#matrix_nginx_proxy_enabled: false
#matrix_ssl_retrieval_method: self-signed
matrix_domain: <redacted>
matrix_ssl_lets_encrypt_support_email: <redacted>
matrix_coturn_turn_static_auth_secret: <redacted>
matrix_synapse_macaroon_secret_key: <redacted>
matrix_homeserver_generic_secret_key: "{{ matrix_synapse_macaroon_secret_key }}"
matrix_element_web_themes_enabled: true

matrix_dimension_enabled: true

matrix_dimension_access_token: <redacted>
matrix_dimension_admins:
  - "<redacted>"
  - "<redacted>"

matrix_nginx_proxy_base_domain_serving_enabled: true
matrix_nginx_proxy_base_domain_homepage_enabled: true

matrix_jitsi_enabled: false

matrix_homeserver_implementation: synapse
matrix_bot_chatgpt_enabled: true
matrix_bot_chatgpt_openai_api_key: <redacted>
matrix_bot_chatgpt_matrix_access_token: <redacted>
```

**Expected behavior**

I have a non-dockerized nginx-server running and used to successfully connect via reverse proxy, prior to new configuration role.

**Matrix Server:**
 - OS: Ubuntu 22.10
 - Architecture amd64

**Comment by spantaleev** [comment]
Should be fixed by 6cda711c0b74ccb6ec192

**Comment by Bergruebe** [comment]
I have now the same problem with the matrix-appservice-discord:
```
matrix-appservice-discord[21163]: docker: Invalid ip address 127.0.0.1:: address 127.0.0.1::: too many colons in address.
```

I looked into you commit https://github.com/spantaleev/matrix-docker-ansible-deploy/commit/6cda711c0b74ccb6ec192cb2a5947d4b27e39d29, there you changed the settings for some app services, but not for the app service discord.

<details>
<summary> My vars.yaml </summary>

```yaml
matrix_domain: XX.tld

# This is something which is provided to Let's Encrypt when retrieving SSL certificates for domains.
#
# In case SSL renewal fails at some point, you'll also get an email notification there.
#
# If you decide to use another method for managing SSL certifites (different than the default Let's Encrypt),
# you won't be required to define this variable (see `docs/configuring-playbook-ssl-certificates.md`).
#
# Example value: someone@example.com
matrix_ssl_lets_encrypt_support_email: 'XX@XXX.tld'

# A shared secret (between Coturn and Synapse) used for authentication.
# You can put any string here, but generating a strong one is preferred (e.g. `pwgen -s 64 1`).
matrix_coturn_turn_static_auth_secret: 'XXX'

# A secret used to protect access keys issued by the server.
# You can put any string here, but generating a strong one is preferred (e.g. `pwgen -s 64 1`).
matrix_synapse_macaroon_secret_key: 'XXX'

# A Postgres password to use for the superuser Postgres user (called `matrix` by default).
#
# The playbook creates additional Postgres users and databases (one for each enabled service)
# using this superuser account.
devture_postgres_connection_password: 'XXX'


# Aktiviere Synapse-Admin
# https://github.com/spantaleev/matrix-docker-ansible-deploy/blob/master/docs/configuring-playbook-synapse-admin.md
matrix_synapse_admin_enabled: true


# nginx deaktivieren
# https://github.com/spantaleev/matrix-docker-ansible-deploy/blob/master/docs/configuring-playbook-own-webserver.md#method-2-fronting-the-integrated-nginx-reverse-proxy-webserver-with-another-reverse-proxy
matrix_playbook_reverse_proxy_type: other-nginx-non-container

# weitere Zertifikate
# https://github.com/spantaleev/matrix-docker-ansible-deploy/blob/master/docs/configuring-playbook-ssl-certificates.md#obtaining-ssl-certificates-for-additional-domains
matrix_ssl_additional_domains_to_obtain_certificates_for:
  - 'XXX'

# matrix-regestrierung
# https://github.com/spantaleev/matrix-docker-ansible-deploy/blob/master/docs/configuring-playbook-matrix-registration.md
matrix_registration_enabled: true

# Generate a strong secret using: `pwgen -s 64 1`.
matrix_registration_admin_secret: "XXX"

# Discord Bot
matrix_appservice_discord_enabled: true
matrix_appservice_discord_client_id: "XXX"
matrix_appservice_discord_bot_token: "XXXX"


# Docker nicht mitinstallieren
matrix_playbook_docker_installation_enabled: false

# Whatsapp Bot
matrix_mautrix_whatsapp_enabled: true

matrix_mautrix_whatsapp_configuration_extension_yaml: |
  permissions:
    "@XX:XX.tld": admin
matrix_mautrix_whatsapp_login_shared_secret: XXX

# https://github.com/spantaleev/matrix-docker-ansible-deploy/blob/master/docs/configuring-playbook-shared-secret-auth.md
# Shared Secret Auth password provider module 
matrix_synapse_ext_password_provider_shared_secret_auth_enabled: true
matrix_synapse_ext_password_provider_shared_secret_auth_shared_secret: XXX

# Mautrix Signal
# https://github.com/spantaleev/matrix-docker-ansible-deploy/blob/master/docs/configuring-playbook-bridge-mautrix-signal.md
matrix_mautrix_signal_enabled: true

matrix_mautrix_signal_login_shared_secret: XXX'

matrix_mautrix_signal_configuration_extension_yaml: |
  permissions:
    "@XX:XX.tld": "admin"

# Element Config
# https://github.com/spantaleev/matrix-docker-ansible-deploy/blob/master/roles/matrix-client-element/defaults/main.yml
matrix_client_element_default_theme: 'theme-dark'


# Synaps Mitteilungen
# https://github.com/spantaleev/matrix-docker-ansible-deploy/blob/master/roles/matrix-synapse/defaults/main.yml#L567-L582
matrix_synapse_configuration_extension_yaml: |
   server_notices:
     system_mxid_localpart: server
     system_mxid_display_name: "Server Mitteilungen"
     system_mxid_avatar_url: "XXX"
     room_name: "Server Mitteilungen"

# Mautrix Telegram
# https://github.com/spantaleev/matrix-docker-ansible-deploy/blob/master/docs/configuring-playbook-bridge-mautrix-telegram.md
matrix_mautrix_telegram_enabled: true
matrix_mautrix_telegram_api_id: XXX
matrix_mautrix_telegram_api_hash: XXX
matrix_mautrix_telegram_bot_token: XXX


matrix_homeserver_generic_secret_key: "{{ matrix_synapse_macaroon_secret_key }}"
```

</details>

**Comment by spantaleev** [comment]
`appservice-discord`, `appservice-slack` and `appservice-webhooks` were fixed in 5e7f30a129077004
